src/appendix/address.md

```diff
@@ -4,6 +4,8 @@ title: "Address"
 
 A Filecoin address is an identifier that refers to an actor in the Filecoin state. All actors (miner actors, the storage market actor, account actors) have an address. This address encodes information about the network to which an actor belongs, the specific type of address encoding, the address payload itself, and a checksum. The goal of this format is to provide a robust address format that is both easy to use and resistant to errors.
 
+Note that each `ActorAddress` in the protocol contains a unique `ActorID` given to it by the `InitActor`. Throughout the protocol, actors are referenced by their IDs rather than full addresses for brevity. However, addresses and IDs are accessible in constant-time using the `AddressMap` (to go from ID to addr) and `IDMap` (in oppositee direction). Using the `IDMap` in conjunction with address resolvers should enable easy lookup of any actor's public keys (should they exist).
```

Comment:
typo: `opposite direction`

Reply:
This sounds confused and some bits contradict my understanding and the implementations.
- Almost universally, actors are referenced by ID-address, not ID. We could change this but the observation isn't true. E.g. state tree keys are ID-addresses and the runtime always resolves to ID addresses before exposing them to actor code.
- ID-addresses are computable from IDs, and vice versa.
- Most actors have an alternative address which is *not* their key in the state tree, but is resolved to that address during message processing. Accounts have a key-based address, and non-singleton actors have a temporary reorg-proof address.
- An account actor's crypto-address (for signature verification) is found by looking up its actor state, keyed by the canonical ID-address. If you already had the alternative pubkey address, there's nothing to look up. There is no map from ID-address to pubkey address because we don't need one.

In order to be informative I think this would need to be a much more detailed description of the various address types, why some actors have multiple, the state tree keys etc.

Reply:
Fair enough. I fixed to include your comments @anorth. Clearly I'm not quite paged in here. Where can I pull in the full details from? Lotus or go-fil (eg)?